Add governance event and governance store

Sotatek-HuyLe3a opened the pull request:
#121,  #100

satran004 commented on stores/governance/src/main/resources/db/store/h2/V0_1100_1__init.sql:
```diff
@@ -0,0 +1,46 @@
+    description     varchar,
+    block           bigint,
+    block_time      bigint,
+    update_datetime timestamp,
```
To handle rollback, we need a slot column.

satran004 commented on stores/governance/src/main/resources/db/store/h2/V0_1100_1__init.sql:
```diff
@@ -0,0 +1,46 @@
+    anchor_hash        varchar(64),
+    block              bigint,
+    block_time         bigint,
+    update_datetime    timestamp,
```
Missing slot column.

satran004 reviewed (commented):
@Sotatek-HuyLe3a Thanks. Looks good to me.
Few minor comments:
1. We need to add slot columns to entities to handle rollback.
2. Add StorageFinder interface & implementation to add few required finders like findByTxHash etc.

Sotatek-HuyLe3a commented on stores/governance/src/main/java/com/bloxbean/cardano/yaci/store/governance/storage/impl/model/GovActionProposalEntity.java:
```diff
@@ -0,0 +1,48 @@
+
+    @Column(name = "anchor_hash")
+    private String anchorHash;
+
```
Right, I'm missing handling rollback event. I will add

Sotatek-HuyLe3a commented on stores/governance/src/main/java/com/bloxbean/cardano/yaci/store/governance/processor/GovActionProposalProcessor.java:
```diff
@@ -0,0 +1,71 @@
+                govActionProposal.setTxHash(txHash);
+                govActionProposal.setIndex(index++);
+                govActionProposal.setType(proposalProcedure.getGovAction().getType());
+                govActionProposal.setDescription(proposalProcedure.getGovAction().toString());
```
I think if we add description field, we could more clearly describe the content of governance action. For example, if gov_type = PARAMETER_CHANGE_ACTION, description field will contain ProtocolParam update info. I am not sure we should add this field or not but when I refer cardano-db-sync, I think this field may be useful. https://github.com/IntersectMBO/cardano-db-sync/blob/master/doc/schema.md#gov_action_proposal

satran004 replied:
Yes, JSON format may be useful and application can parse it if required. Let's try with JSON type field for description. Should we call it description or body/details? Because this is more than simple description :)

satran004 replied:
@Sotatek-HuyLe3a

Sotatek-HuyLe3a replied:
Yes, i will try it and change field name

Sotatek-HuyLe3a commented:
> @Sotatek-HuyLe3a Thanks. Looks good to me. Few minor comments:
> 
> 1. We need to add slot columns to entities to handle rollback.
> 2. Add StorageFinder interface & implementation to add few required finders like findByTxHash etc.

@satran004 , I updated code after your first review. I  also added the governance-api module and tests for the governance store. Please review again, Thanks !

satran004 reviewed (approved):
@Sotatek-HuyLe3a Thanks for updating the PR.
Looks good to me.

Can you please add the new governance api endpoints to this test script before merging? As there is no data available for test on preprod, just keep the placeholder for parameters for now.

https://github.com/bloxbean/yaci-store/blob/main/test-scripts/query.http

Sotatek-HuyLe3a commented:
> @Sotatek-HuyLe3a Thanks for updating the PR. Looks good to me.
> 
> Can you please add the new governance api endpoints to this test script before merging? As there is no data available for test on preprod, just keep the placeholder for parameters for now.
> 
> https://github.com/bloxbean/yaci-store/blob/main/test-scripts/query.http

I just updated @satran004

Sotatek-HuyLe3a commented:
> @Sotatek-HuyLe3a Thanks for updating the PR. Looks good to me.
> 
> Can you please add the new governance api endpoints to this test script before merging? As there is no data available for test on preprod, just keep the placeholder for parameters for now.
> 
> https://github.com/bloxbean/yaci-store/blob/main/test-scripts/query.http

I just updated @satran004
